plugins/flytekit-duckdb/flytekitplugins/duckdb/task.py

```diff
@@ -21,19 +42,53 @@ class DuckDBQuery(PythonInstanceTask):
     def __init__(
         self,
         name: str,
-        query: Union[str, List[str]],
+        query: Optional[Union[str, List[str]]] = None,
         inputs: Optional[Dict[str, Union[StructuredDataset, list]]] = None,
+        provider: Union[DuckDBProvider, Callable] = DuckDBProvider.LOCAL,
         **kwargs,
     ):
         """
         This method initializes the DuckDBQuery.
 
+        Note that the provider can be one of the default providers listed in DuckDBProvider or a custom callable like the following:
+
+        def custom_connect_motherduck(token: str):
+            return duckdb.connect("md:", config={"motherduck_token": token, "another_config": "hello"})
+
+        DuckDBQuery(..., provider=custom_connect_motherduck)
+
+        Also note that a query can be provided at runtime if query=None is provided.
+
+        duckdb_query = DuckDBQuery(
```

Comment:
since the inputs must be provided at registration time, does it make sense to provide the query at runtime?

Reply:
The actual contents of the query could differ at runtime based on some logic in say a dynamic task. If the input type is a dataframe, you can still query it in different ways.